common/components/prefix_sum.hpp.inc

```diff
@@ -130,8 +131,10 @@ __global__ __launch_bounds__(block_size) void start_prefix_sum(
     }
 
     if (element_id == 0) {
-        // Store the total sum
-        block_sum[blockIdx.x] = prefix_helper[block_size - 1];
+        // Store the total sum except the last block
+        if (blockIdx.x + 1 < gridDim.x) {
+            block_sum[blockIdx.x] = prefix_helper[block_size - 1];
+        }
```

Comment:
Could you also change the documentation of `block_sum` to have at least `ceildiv(num_elements, block_size) - 1` since you never access `block_size - 1`.